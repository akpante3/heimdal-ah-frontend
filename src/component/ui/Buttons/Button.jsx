import React from 'react';
import PropTypes from 'prop-types';
import './Style.scss';
/**
 * @description - Heimdal Authors Haven Button Component
 * @param {props} Class - the class of the button(bootstrap or custom)
 * @param {props} onClick - the function to be performed by the button when it is clicked
 * @param {props} label - the title of the button
 * @param {props} type - the type of the button(signin,signup,login,follow,tag,login2,signup2,
 * subscribe,publish,report,notifications,article,save-profile,cancel,edit-profile,reply,comment,
 * stories,facebook,twitter,google)
 * @returns {component} Button
 */
const Button = (props) => {
  const {
    Class, onClick, label, type,
  } = props;

  if (type) {
    return (
      <button type="submit" className={`button btn-${type}  ${Class}`} onClick={onClick}>
        {label}
      </button>
    );
  }
  return (
    <button type="button" className={`button btn-${type}`} onClick={onClick}>
      {label}
    </button>
  );
};

Button.defaultProps = {
  Class: '',
  type: 'default',
  onClick: () => {},
  label: 'Default',
};

Button.propTypes = {
  Class: PropTypes.string,
  onClick: PropTypes.func,
  label: PropTypes.string,
  type: PropTypes.oneOf([
    'signin',
    'signup',
    'login',
    'follow',
    'tag',
    'login2',
    'signup2',
    'subscribe',
    'publish',
    'report',
    'notifications',
    'article',
    'save-profile',
    'cancel',
    'edit-profile',
    'reply',
    'comment',
    'stories',
    'facebook',
    'twitter',
    'google',
    'default',
  ]),
};

export default Button;
