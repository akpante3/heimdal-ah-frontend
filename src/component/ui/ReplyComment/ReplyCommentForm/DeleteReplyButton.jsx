import React from 'react';
import FontAwesome from 'react-fontawesome';
import { connect } from 'react-redux';


const DeleteButton = (props) => {
  const deleteComment = (e) => {
    e.preventDefault();
    const { commentId } = props;
    props.deleteArticleComment(1, commentId);
  };
  return (
    <button className='archive_button btn btn-danger' onClick={e => deleteComment(e)} type='submit'><FontAwesome name='trash' className='' /></button>
  );
};

const mapStateToProps = state => ({
  comment: state.comment,
});

export default connect(mapStateToProps, { })(DeleteButton);
