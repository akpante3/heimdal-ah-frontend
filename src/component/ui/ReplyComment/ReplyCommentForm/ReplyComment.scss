
.reply_button {
  cursor: pointer;
}
.reply_form {
  margin-top: 20px;
}

.reply_input {
  display: flex;
  justify-content: flex-end;
}

.btn__reply {
  float: right;
}

.error_msg {
  margin-top: 10px;
  text-align: end;
  margin-left: 50px;
}