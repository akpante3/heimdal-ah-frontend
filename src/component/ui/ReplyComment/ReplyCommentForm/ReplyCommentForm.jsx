/* eslint-disable react/forbid-prop-types */
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { FormGroup, FormControl } from 'react-bootstrap';
import { connect } from 'react-redux';
import './ReplyComment.scss';
import Button from '../../Buttons/Button';
import Alert from '../../Alert/Alert';
import { postCommentReplies } from '../../../../actions/ReplyActions/ReplyAction';


/**
  * renderComponent
  * @method Class Class based Component
  * @summary React component for rendering the card components
  * @param {object}  Replies contains the Replies of a comment
  * @param {Object} props - React PropTypes
  * @property {String|Object} className - String className compatible object for styling
  * @return {Node} React node containing comment form view
  */
export class ReplyCommentForm extends Component {
  state = {
    content: '',
    replyError: '',
  }

  onChange = (e) => {
    this.setState({
      [e.target.name]: e.target.value,
      replyError: '',
    });
  }

  onBlur = (e) => {
    if (e.target.value === '') {
      this.setState({
        replyError: 'Reply Content cannot be Empty',
      });
    } else {
      this.setState({
        formIsValid: true,
        replyError: '',
      });
    }
  }

  submitForm = (e) => {
    e.preventDefault();
    const { user } = this.props.auth;
    const { commentId } = this.props;
    const { formIsValid, content } = this.state;
    if (content.length >= 1500) {
      this.setState({
        formIsValid: false,
        replyError: "Can't have more than 1500 characters",
      });
    } else if (formIsValid) {
      const data = { content };
      this.props.postCommentReplies(commentId, data, user.image);
      this.setState({
        content: '',
      });
    } else {
      this.setState({
        replyError: 'Reply Content cannot be Empty',
      });
    }
  }

  /**
     * Renders the component.
     *
     * @memberof app.components.Replies
     * @return {string} - HTML markup for the component
   */
  render() {
    const { content, replyError } = this.state;
    return (
      <div>
        <div className='error_msg'>
          {' '}
          {replyError !== '' && <Alert type='warning' message={this.state.replyError} title='Invalid Credenntials' /> }
        </div>
        <div className='reply_input'>
          <form className='' onSubmit={this.submitForm}>
            <FormGroup controlId="formControlsTextarea">
              <FormControl
                onBlur={this.onBlur}
                onChange={this.onChange}
                componentClass="textarea"
                value={content}
                name='content'
                className='reply_form'
                rows="5"
                cols="80"
                placeholder="textarea"
              />
            </FormGroup>
            <Button type="reply" Class='btn__reply' label="Add Reply" />
          </form>
        </div>
      </div>
    );
  }
}

ReplyCommentForm.propTypes = {
  auth: PropTypes.object.isRequired,
};
/**
 * @method module:Reactator.ReduxContainerBuilderMapStateToProps
 *
 * @param {function} mapStateToProps - function mapping redux state to props
 *
 * @return {ReduxContainerBuilder} this builder
*/
const mapStateToProps = state => ({
  auth: state.auth,
});

export default connect(mapStateToProps, { postCommentReplies })(ReplyCommentForm);
