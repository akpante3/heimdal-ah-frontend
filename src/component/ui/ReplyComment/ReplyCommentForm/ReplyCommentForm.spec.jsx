/* eslint-disable no-undef */
/* eslint-disable react/react-in-jsx-scope */
import { shallow } from 'enzyme';
import { ReplyCommentForm } from './ReplyCommentForm';
import Button from '../../Buttons/Button';


describe('rendering', () => {
  let wrapper;
  let props;
  beforeEach(() => {
    props = {
      postCommentReplies: jest.fn(),
      auth: '',
    };
  });

  it('renders One Reply Form components', () => {
    wrapper = shallow(<ReplyCommentForm {...props} />);
    expect(wrapper.find(Button)).toEqual({});
  });

  it('should render the component', () => {
    wrapper = shallow(<ReplyCommentForm {...props} />);
    expect(wrapper.length).toBe(1);
  });

  it('should render the component', () => {
    wrapper = shallow(<ReplyCommentForm {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
