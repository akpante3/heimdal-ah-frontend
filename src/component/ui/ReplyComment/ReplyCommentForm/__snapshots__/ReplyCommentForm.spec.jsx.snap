// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering should render the component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ReplyCommentForm
    auth=""
    postCommentReplies={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="error_msg"
        >
           
        </div>,
        <div
          className="reply_input"
        >
          <form
            className=""
            onSubmit={[Function]}
          >
            <FormGroup
              bsClass="form-group"
              controlId="formControlsTextarea"
            >
              <FormControl
                bsClass="form-control"
                className="reply_form"
                cols="80"
                componentClass="textarea"
                name="content"
                onBlur={[Function]}
                onChange={[Function]}
                placeholder="textarea"
                rows="5"
                value=""
              />
            </FormGroup>
            <Button
              Class="btn__reply"
              label="Add Reply"
              onClick={[Function]}
              type="reply"
            />
          </form>
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            " ",
            false,
          ],
          "className": "error_msg",
        },
        "ref": null,
        "rendered": Array [
          " ",
          false,
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <form
            className=""
            onSubmit={[Function]}
          >
            <FormGroup
              bsClass="form-group"
              controlId="formControlsTextarea"
            >
              <FormControl
                bsClass="form-control"
                className="reply_form"
                cols="80"
                componentClass="textarea"
                name="content"
                onBlur={[Function]}
                onChange={[Function]}
                placeholder="textarea"
                rows="5"
                value=""
              />
            </FormGroup>
            <Button
              Class="btn__reply"
              label="Add Reply"
              onClick={[Function]}
              type="reply"
            />
          </form>,
          "className": "reply_input",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <FormGroup
                bsClass="form-group"
                controlId="formControlsTextarea"
              >
                <FormControl
                  bsClass="form-control"
                  className="reply_form"
                  cols="80"
                  componentClass="textarea"
                  name="content"
                  onBlur={[Function]}
                  onChange={[Function]}
                  placeholder="textarea"
                  rows="5"
                  value=""
                />
              </FormGroup>,
              <Button
                Class="btn__reply"
                label="Add Reply"
                onClick={[Function]}
                type="reply"
              />,
            ],
            "className": "",
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "form-group",
                "children": <FormControl
                  bsClass="form-control"
                  className="reply_form"
                  cols="80"
                  componentClass="textarea"
                  name="content"
                  onBlur={[Function]}
                  onChange={[Function]}
                  placeholder="textarea"
                  rows="5"
                  value=""
                />,
                "controlId": "formControlsTextarea",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "bsClass": "form-control",
                  "className": "reply_form",
                  "cols": "80",
                  "componentClass": "textarea",
                  "name": "content",
                  "onBlur": [Function],
                  "onChange": [Function],
                  "placeholder": "textarea",
                  "rows": "5",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "Class": "btn__reply",
                "label": "Add Reply",
                "onClick": [Function],
                "type": "reply",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "form",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="error_msg"
          >
             
          </div>,
          <div
            className="reply_input"
          >
            <form
              className=""
              onSubmit={[Function]}
            >
              <FormGroup
                bsClass="form-group"
                controlId="formControlsTextarea"
              >
                <FormControl
                  bsClass="form-control"
                  className="reply_form"
                  cols="80"
                  componentClass="textarea"
                  name="content"
                  onBlur={[Function]}
                  onChange={[Function]}
                  placeholder="textarea"
                  rows="5"
                  value=""
                />
              </FormGroup>
              <Button
                Class="btn__reply"
                label="Add Reply"
                onClick={[Function]}
                type="reply"
              />
            </form>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              " ",
              false,
            ],
            "className": "error_msg",
          },
          "ref": null,
          "rendered": Array [
            " ",
            false,
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <form
              className=""
              onSubmit={[Function]}
            >
              <FormGroup
                bsClass="form-group"
                controlId="formControlsTextarea"
              >
                <FormControl
                  bsClass="form-control"
                  className="reply_form"
                  cols="80"
                  componentClass="textarea"
                  name="content"
                  onBlur={[Function]}
                  onChange={[Function]}
                  placeholder="textarea"
                  rows="5"
                  value=""
                />
              </FormGroup>
              <Button
                Class="btn__reply"
                label="Add Reply"
                onClick={[Function]}
                type="reply"
              />
            </form>,
            "className": "reply_input",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <FormGroup
                  bsClass="form-group"
                  controlId="formControlsTextarea"
                >
                  <FormControl
                    bsClass="form-control"
                    className="reply_form"
                    cols="80"
                    componentClass="textarea"
                    name="content"
                    onBlur={[Function]}
                    onChange={[Function]}
                    placeholder="textarea"
                    rows="5"
                    value=""
                  />
                </FormGroup>,
                <Button
                  Class="btn__reply"
                  label="Add Reply"
                  onClick={[Function]}
                  type="reply"
                />,
              ],
              "className": "",
              "onSubmit": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "bsClass": "form-group",
                  "children": <FormControl
                    bsClass="form-control"
                    className="reply_form"
                    cols="80"
                    componentClass="textarea"
                    name="content"
                    onBlur={[Function]}
                    onChange={[Function]}
                    placeholder="textarea"
                    rows="5"
                    value=""
                  />,
                  "controlId": "formControlsTextarea",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "bsClass": "form-control",
                    "className": "reply_form",
                    "cols": "80",
                    "componentClass": "textarea",
                    "name": "content",
                    "onBlur": [Function],
                    "onChange": [Function],
                    "placeholder": "textarea",
                    "rows": "5",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "Class": "btn__reply",
                  "label": "Add Reply",
                  "onClick": [Function],
                  "type": "reply",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "form",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
